sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -303,34 +307,130 @@ export class ManagementClient extends LinkEntity {
 
-      if (!options) {
-        options = {};
-      }
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            let count = 0;
+
+            const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
+            let timeTakenByInit = 0;
+
+            const rejectOnAbort = () => {
+              const requestName = options.requestName;
+              const desc: string =
+                `[${this._context.connectionId}] The request "${requestName}" ` +
+                `to has been cancelled by the user.`;
+              log.error(desc);
+              const error = new AbortError(
+                `The ${
+                  requestName ? requestName + " " : ""
+                }operation has been cancelled by the user.`
+              );
+
+              reject(error);
+            };
+
+            if (aborter) {
+              if (aborter.aborted) {
+                return rejectOnAbort();
+              }
+            }
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
+              log.mgmt(
+                "[%s] Acquiring lock to get the management req res link.",
+                this._context.connectionId
+              );
+
+              const initOperationStartTime = Date.now();
+
+              const actionAfterTimeout = () => {
+                const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
+                const e: Error = {
+                  name: "OperationTimeoutError",
+                  message: desc
+                };
+
+                return reject(translate(e));
+              };
+
+              const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
+
+              try {
+                await defaultLock.acquire(this.managementLock, () => {
+                  return this._init();
+                });
+              } catch (err) {
+                return reject(translate(err));
+              } finally {
+                clearTimeout(waitTimer);
+              }
+              timeTakenByInit = Date.now() - initOperationStartTime;
+            }
+
+            const remainingOperationTimeoutInMs = retryTimeoutInMs - timeTakenByInit;
+
+            const sendRequestOptions: SendRequestOptions = {
+              abortSignal: options.abortSignal,
+              requestName: options.requestName,
+              timeoutInSeconds: remainingOperationTimeoutInMs / 1000
+            };
+
+            count++;
+            if (count !== 1) {
+              // Generate a new message_id every time after the first attempt
+              request.message_id = generate_uuid();
+            } else if (!request.message_id) {
+              // Set the message_id in the first attempt only if it is not set
+              request.message_id = generate_uuid();
+            }
+
+            if (this._mgmtReqResLink) {
```

Comment:
You shouldn't need this check here since you know at this point that you either have a link already, or you just created one. Since you don't have any other non-blocking code before reaching this point it's safe to assume it exists.

If you wanted to keep this check here, we should probably throw an error if it doesn't exist rather than resolve with an undefined value, but again I don't think this check is necessary.

Reply:
Right, thought I specifically updated this - looks like the change got missed out